sympy/functions/elementary/exponential.py

```diff
@@ -436,11 +436,27 @@ def _eval_nseries(self, x, n, logx):
         if arg0 in [-oo, oo]:
             return self
         t = Dummy("t")
-        exp_series = exp(t)._taylor(t, n)
+        nterms = n
+        try:
+            cf = Order(arg.as_leading_term(x), x).getn()
+        except NotImplementedError:
+            cf = 0
+        if cf > 0:
+            try:
+                nterms = (n/cf).ceiling()
+            except (AttributeError, TypeError):
+                pass
+        exp_series = exp(t)._taylor(t, n, nterms)
         o = exp_series.getO()
         exp_series = exp_series.removeO()
         r = exp(arg0)*exp_series.subs(t, arg_series - arg0)
-        r += Order(o.expr.subs(t, (arg_series - arg0)), x)
+        if cf != 0:
+            try:
+                r += Order(o.expr.subs(t, (arg_series - arg0)), x)/x**((cf-1)*n)
+            except TypeError:
```

Comment:
Is this still needed?

Reply:
Maybe we can Its logically right also and I have also checked some examples.